### PR TITLE
SelfSubjectAccessReview auth test coverage

### PR DESCRIPTION
**Pull request by Skyterix1991**

### CODE REVIEW

**Comment by rohanKanojia** [comment]
ok to test

File: kubernetes-tests/src/test/java/io/fabric8/kubernetes/client/mock/SelfSubjectAccessReviewAuthTest.java [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestMethodOrder;
+import org.junit.jupiter.migrationsupport.rules.EnableRuleMigrationSupport;
+
+import java.io.IOException;
+import java.nio.charset.Charset;
+
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
```
Comment (kunal-kushwaha) [inline kubernetes-tests/src/test/java/io/fabric8/kubernetes/client/mock/SelfSubjectAccessReviewAuthTest.java]: Import assertTrue as well.

File: kubernetes-tests/src/test/java/io/fabric8/kubernetes/client/mock/SelfSubjectAccessReviewAuthTest.java [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+    @Rule
+    public KubernetesServer server = new KubernetesServer();
+
+    @Test
+    @Order(1)
+    public void testList() {
+        KubernetesClient client = server.getClient();
+
+        initializeCreateRequestHandler();
```
Comment (manusa) [inline kubernetes-tests/src/test/java/io/fabric8/kubernetes/client/mock/SelfSubjectAccessReviewAuthTest.java]: Method invoked for every test. Probably better in a setUp method ( `@BeforeEach`)

**Comment by rohanKanojia** [comment]
retest this please

**Comment by rohanKanojia** [comment]
retest this please

File: kubernetes-tests/src/test/java/io/fabric8/kubernetes/client/mock/SelfSubjectAccessReviewAuthTest.java [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+        server.expect().post().withPath("/apis/authorization.k8s.io/v1/namespaces/test/selfsubjectaccessreviews").andReply(200, recordedRequest -> {
+            ObjectMapper mapper = new ObjectMapper();
+
+            SelfSubjectAccessReview reviewResponse = null;
+
+            try {
+                reviewResponse = mapper.readValue(recordedRequest.getBody().readString(Charset.defaultCharset()), SelfSubjectAccessReview.class);
+                reviewResponse.setStatus(new SubjectAccessReviewStatus(true, false, "",""));
+            } catch (IOException e) {
+                e.printStackTrace();
```
Comment (manusa) [inline kubernetes-tests/src/test/java/io/fabric8/kubernetes/client/mock/SelfSubjectAccessReviewAuthTest.java]: Test should fail if there is an exception, add `throws IOException` to method signature and remove try-catch block.

**Pull request closed without merging**